src/olympia/amo/utils.py

```diff
@@ -641,11 +641,17 @@ def resize_image(source, destination, size=None):
         original_size = im.size
         if size:
             im = processors.scale_and_crop(im, size)
-    with storage.open(destination, 'wb') as fp:
-        # Save the image to PNG in destination file path. Don't keep the ICC
-        # profile as it can mess up pngcrush badly (mozilla/addons/issues/697).
-        im.save(fp, 'png', icc_profile=None)
-    pngcrush_image(destination)
+    if not format:
```

Comment:
this will actually end up with all images being saved as png again because when `format='jpeg'` it's not, `not format`.  When `format == 'png'` you want to do something special (specify the iccprofile and pngcrush), else you want to just convert and save with the format specified.

Reply:
That makes sense now. Apologies for the oversight!!